### ISSUE TITLE
There was a problem getting the parent element.

### ISSUE THREAD

**Issue by siebird** [spicywebau/craft-neo#531]
Having issues accessing an entry after saving. Initially, I thought it was every page after being saved. However, it seems to be specific to an entry. 

Not sure if its related to Neo or Craft. https://github.com/craftcms/cms/issues/10327

Any utilities to repair this entry?

**Comment by ttempleton** [comment]
If it's related to Neo, it'd likely be something going wrong when trying to merge Neo block structures from a saved entry into one of its drafts, or into a provisional draft if another draft has just been applied — could you please provide a stack trace as that should confirm if block structure merging is causing it. I'll also keep an eye on craftcms/cms#10327.

**Comment by siebird** [comment]
Let me know if this is what you are looking for:
```
2022-01-07 09:19:32 [-][1][l3l3tm8n6c5s4u40la3p48eu2h][error][yii\base\Exception] yii\base\Exception: There was a problem getting the parent element. in /home/forge/domain.com/vendor/craftcms/cms/src/services/Structures.php:292
Stack trace:
#0 /home/forge/domain.com/vendor/spicyweb/craft-neo/src/services/Fields.php(618): craft\services\Structures->prepend()
#1 /home/forge/domain.com/vendor/spicyweb/craft-neo/src/Field.php(717): benf\neo\services\Fields->mergeCanonicalChanges()
#2 /home/forge/domain.com/vendor/craftcms/cms/src/base/Element.php(4146): benf\neo\Field->afterElementPropagate()
#3 /home/forge/domain.com/vendor/craftcms/cms/src/elements/Entry.php(1722): craft\base\Element->afterPropagate()
#4 /home/forge/domain.com/vendor/craftcms/cms/src/services/Elements.php(842): craft\elements\Entry->afterPropagate()
#5 [internal function]: craft\services\Elements->craft\services\{closure}()
#6 /home/forge/domain.com/vendor/yiisoft/yii2/db/Connection.php(812): call_user_func()
#7 /home/forge/domain.com/vendor/craftcms/cms/src/services/Elements.php(843): yii\db\Connection->transaction()
#8 /home/forge/domain.com/vendor/craftcms/cms/src/controllers/EntriesController.php(700): craft\services\Elements->mergeCanonicalChanges()
#9 /home/forge/domain.com/vendor/craftcms/cms/src/controllers/EntriesController.php(86): craft\controllers\EntriesController->_prepEditEntryVariables()
#10 [internal function]: craft\controllers\EntriesController->actionEditEntry()
#11 /home/forge/domain.com/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#12 /home/forge/domain.com/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#13 /home/forge/domain.com/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#14 /home/forge/domain.com/vendor/craftcms/cms/src/web/Application.php(287): yii\base\Module->runAction()
#15 /home/forge/domain.com/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction()
#16 /home/forge/domain.com/vendor/craftcms/cms/src/web/Application.php(272): yii\web\Application->handleRequest()
#17 /home/forge/domain.com/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest()
#18 /home/forge/domain.com/public/index.php(26): yii\base\Application->run()
#19 {main}
2022-01-07 09:19:32 [-][1][l3l3tm8n6c5s4u40la3p48eu2h][info][application] $_GET = [
    'site' => 'primary'
]
```

**Comment by ttempleton** [comment]
Thanks! Looks like that is coming from the block structure merging. Could you please send your composer.json/lock and database backup to plugins@spicyweb.com.au and let us know in the email which entry this is happening with, and we'll have a look.

**Comment by ttempleton** [comment]
Thanks for sending that through. It turned out that the block structure merging wasn't skipping blocks that had been deleted in the provisional draft, which caused this error in this case with a parent block that had been deleted. Fixed now in Neo 2.12.2.

**Comment by siebird** [comment]
Thanks for the quick fix. We appreciate it!
